releases/release-5.0.2.md

```diff
@@ -0,0 +1,114 @@
+---
+title: TiDB 5.0.2 Release Notes
+---
+
+# TiDB 5.0.2 Release Notes
+
+Release date: June 9, 2021
+
+TiDB version: 5.0.2
+
+## Compatibility Changes
+
+## New Features
+
+## Improvements
+
++ TiDB
+
+    - Skip reading mysql.stats_histograms if cached stats is up-to-date [#24317](https://github.com/pingcap/tidb/pull/24317)
+
++ TiFlash
+
+    - Apply Region snapshots by ingesting files to greatly reduce the memory usage [#1867](https://github.com/pingcap/tics/pull/1867)
+    - Optimize the table lock to avoid reading and DDL jobs from blocking each other [#1858](https://github.com/pingcap/tics/pull/1858)
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix index join panic on prefix index on some cases [#24824](https://github.com/pingcap/tidb/pull/24824)
+    - Fix the issue point get cached plan of prepared statement is incorrectly used by in transaction point get statement. [#24765](https://github.com/pingcap/tidb/pull/24765)
+    - Fix write wrong prefix index value when collation is ascii_bin/latin1_bin [#24680](https://github.com/pingcap/tidb/pull/24680)
+    - Fix the issue that ongoing transaction could be aborted by the gc worker. [#24652](https://github.com/pingcap/tidb/pull/24652)
+    - Fix bug that point query maybe get wrong on the clustered index when new-collation enabled but new-row-format disabled [#24611](https://github.com/pingcap/tidb/pull/24611)
+    - Refactor converting partition keys for shuffle hash join [#24490](https://github.com/pingcap/tidb/pull/24490)
+    - Fix panic when building plan for a query containing HAVING clause. [#24489](https://github.com/pingcap/tidb/pull/24489)
+    - Planner: fix column pruning bug for Apply and Join [#24437](https://github.com/pingcap/tidb/pull/24437)
+    - Fix the bug that primary lock fallen back from async commit cannot be resolved. [#24397](https://github.com/pingcap/tidb/pull/24397)
+    - Statistics: fix a statistics GC problem that can cause duplicated fm-sketch records [#24357](https://github.com/pingcap/tidb/pull/24357)
+    - Avoid unnecessary pessimistic rollback when pessimistic locking receives ErrKeyExists [#23800](https://github.com/pingcap/tidb/pull/23800)
+    - Fix numeric literals cannot be recognized when set 'ANSI_QUOTES' sql_mode [#25015](https://github.com/pingcap/tidb/pull/25015)
+    - No longer allowing `INSERT INTO table PARTITION (<partitions>) ... ON DUPLICATE KEY UPDATE` to read from non-listed partitions [#25000](https://github.com/pingcap/tidb/pull/25000)
+    - Fix "index out of range" error when a SQL contains both group by and union. [#24551](https://github.com/pingcap/tidb/pull/24551)
+    - Fix wrong collation for concat function [#24301](https://github.com/pingcap/tidb/pull/24301)
+    - Fix global variable collation_server does not take effect in new session [#24156](https://github.com/pingcap/tidb/pull/24156)
+
++ TiFlash
```

Comment:
+ TiFlash

    - Fix the issue of incorrect results when cloning shared delta index concurrently
    - Fix the potential issue that TiFlash fails to restart with incomplete data
    - Fix the issue that old dm files are not removed atomically
    - Fix the potential panic that occurs when feature `compaction filter` is enabled
    - Fix the potential issue that `ExchangeSender` will send duplicated chunks
    - Fix the issue that TiFlash can not resolve fallen back `async commit` lock
    - Fix the issue of incorrect results when the return columns of casting timezone contains timestamp type
    - Fix the issue that TiFlash panics during Segment Split
    - Fix the issue that execution information about non root MPP task is not accurate